thoth/package_extract/image.py

```diff
@@ -525,6 +527,42 @@ def get_image_size(path: str) -> int:
     return total_size
 
 
+def _get_cuda_version(path: str) -> Optional[str]:
+    """Get the cuda version."""
+    output = run_command("{}/usr/local/cuda/version.txt".format(path), timeout=2).stdout
+    lines = output.splitlines()
+    if len(lines) != 3:
+        _LOGGER.debug(
+            "Unable to detect CUDA version from usr/local/cuda/version.txt output: %r",
+            output,
+        )
+    else:
+        return lines[-1]
+
+    result = run_command("nvcc --version", raise_on_error=False)
+    if result.return_code != 0:
+        _LOGGER.info("No CUDA version detected")
+        _LOGGER.debug(
+            "Unable to detect CUDA version - nvcc returned non-zero version: %s",
+            result.to_dict(),
+        )
+        return None
+
+    lines = result.stdout.splitlines()
+    version_info = lines[-1].split(",")
+    if len(version_info) != 3:
+        _LOGGER.debug(
+            "Unable to detect CUDA version from nvcc output: %r", result.stdout
+        )
+        return None
+
+    cuda_version = version_info[1].strip()[len("release "):]
+
+    _LOGGER.info("Detected CUDA version: %r", cuda_version)
+    return cuda_version
+    return None
```

Comment:
Can we return here a dict so we know based on which method we obtained version information?

```python
{
    "/usr/local/cuda/version.txt": ...
    "nvcc_version": ...
}
```

On the syncing side, we will just take a look if versions match and if they are correct (no mismatch).